src/main/java/uk/gov/hmcts/reform/managecase/service/ras/RoleAssignmentService.java

```diff
@@ -39,6 +57,69 @@ public void deleteRoleAssignments(List<RoleAssignmentsDeleteRequest> deleteReque
         }
     }
 
+    public void createCaseRoleAssignments(final List<RoleAssignmentsAddRequest> addRequest) {
+
+        if (addRequest != null && !addRequest.isEmpty()) {
+            final var queryRequests = addRequest.stream()
+                .map(request -> createCaseRoleAssignments(
+                         request.getCaseDetails(),
+                         request.getUserId(),
+                         request.getRoleNames(),
+                         false
+                     )
+                ).collect(Collectors.toList());
+
+            queryRequests.stream().forEach(assignmentRequest ->
+                                           roleAssignmentServiceHelper.createRoleAssignment(assignmentRequest)
+            );
```

Comment:
Looks like we could just use one single stream for this processing
```suggestion
            addRequest.stream()
                .map(request -> createCaseRoleAssignments(
                         request.getCaseDetails(),
                         request.getUserId(),
                         request.getRoleNames(),
                         false
                     )
                ).forEach(assignmentRequest ->
                                           roleAssignmentServiceHelper.createRoleAssignment(assignmentRequest)
            );
```